fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/Type.java

```diff
@@ -1039,6 +1080,12 @@ public static Field of(@Nonnull final Type fieldType, @Nonnull final Optional<St
         @Nullable
         private final Type innerType;
 
+        /**
+         * Memoized hash function.
+         */
+        @Nonnull
+        private final Supplier<Integer> memoizedHashFunction;
```

Comment:
damn that there are no traits in Java

Reply:
That would be really nice to have!
I guess it would've been nice, at least for the planner, to choose Scala as a programming language :-)

Reply:
we can still do that -- laughs in Mr. Burns' voice --